README.md

```diff
@@ -70,6 +71,46 @@ Fedora 25. Other nodesets are currently not present. The module requires at
 least Puppet 4.9.4 (this is the first version with proper data-in-modules
 support).
 
+
+## Reference
+
+The main class provides an interface that can be used. It can be configured via
+the upcoming parameters. All other classes are internal and can't be accessed
+directly.
+
+### `$manage_facts`
+
+This is a boolean, it defaults to true on all systems except for the CentOS
+family (see [limitations](#limitations). If true, the module will provide
+a script to run lldpctl and save the output as json as structured fact. A
+systemd timer or a cronjob will be deployed to execute the script periodically.
+This decouples the fact generation from the actual facter/puppet.
+
+### `$manage_service`
+
+Boolean, if true, the module will enable the service. Defaults to true.
+
+### `$manage_repo`
+
+Boolean, if true, the module will setup the upstream repository. Defaults to
+true on Debian and CentOS family.
+
+### `$manage_jq`
+
+Boolean. jq is a dependency needed for the fact generation. This modules can
+install it. Defaults to true on all systems except for CentOS. See also
+[limitations](#limitations).
+
+### `$repourl`
```

Comment:
If you wanted to be crystal clear you could have called it something like `$repourl_suffix`, but that's for another day.

Reply:
also it isn't a suffix, it is in the middle of the URL